### PR TITLE
Raise statuses with the scheduler

### PR DESCRIPTION
**Pull request by mcoughlin**
**Does this pull request make any changes to the database?**
None

Code changes that affect the database require special attention for data migration.
N/A

### CODE REVIEW

File: growth/too/tasks/scheduler.py [hunk 1]
```diff
@@ -14,8 +14,10 @@
 from . import celery
 from .. import models
 from .. import views
+from flask import flash
 
-ZTF_URL = 'http://tunnel:9999'
+
+ZTF_URL = 'http://localhost:9999'
```
Comment (lpsinger) [inline growth/too/tasks/scheduler.py:20]: This looks like a change for local testing that was included by accident.

Comment (mcoughlin) [reply growth/too/tasks/scheduler.py]: ARGHHHHHHHHHHHHH. Fixed. Thanks Leo.

File: growth/too/tasks/scheduler.py [hunk 2]
```diff
@@ -157,7 +159,14 @@ def schedule_ztf(json_data):
...
                            'queue_type': 'list'})
-    r.raise_for_status()
+
+    if r.status_code == 200:
+        if "already exists" in r.text:
+            flash(r.text, 'danger')
+        else:
+            flash('Successfully submitted', 'success')
+    elif r.status_code == 400:
+        flash('Submission failed... contact the admins.', 'danger')
```
Comment (lpsinger) [inline growth/too/tasks/scheduler.py:169]: Not raising an exception results in the loss of valuable diagnostic information in our logs. I am not sure if it is right to use `flash` here at all. What about just setting Flask's HTTP error status page with contact information for us?

Comment (mcoughlin) [reply growth/too/tasks/scheduler.py:169]: I am not opposed to this, but maybe easiest is @ebellm reporting the issue in the message?

File: growth/too/tasks/scheduler.py [hunk 3]
```diff
@@ -157,7 +159,14 @@ def schedule_ztf(json_data):
                            'validity_window_mjd':
                                json_data["validity_window_mjd"],
                            'queue_type': 'list'})
-    r.raise_for_status()
+
+    if r.status_code == 200:
```
[the remark this answers is not in the document]
Comment (ebellm) [reply growth/too/tasks/scheduler.py:163]: This is a `PUT` request, which you taught me should be idempotent--so wouldn't changing the status code contradict that?  Should I make this a `POST` instead?  

Or should we just break the rules a bit?

File: growth/too/tasks/scheduler.py (hunk 2 above)
Comment (ebellm) [reply growth/too/tasks/scheduler.py:169]: I can definitely pass more error text back in the HTTP response body, which you could then display.  Stay tuned.

File: growth/too/tasks/scheduler.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (lpsinger) [reply growth/too/tasks/scheduler.py:163]: According to https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT:

> If the target resource does not have a current representation and the PUT request successfully creates one, then the origin server must inform the user agent by sending a 201 (Created) response.
> 
>     HTTP/1.1 201 Created
>     Content-Location: /new.html
>
> If the target resource does have a current representation and that representation is successfully modified in accordance with the state of the enclosed representation, then the origin server must send either a 200 (OK) or a 204 (No Content) response to indicate successful completion of the request.
> 
>     HTTP/1.1 204 No Content
>     Content-Location: /existing.html

File: growth/too/tasks/scheduler.py (hunk 2 above)
Comment (lpsinger) [reply growth/too/tasks/scheduler.py:169]: I like that idea. But I do suggest both flashing the message and also catching and logging the exception with `log.exception()`.

File: growth/too/tasks/scheduler.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ebellm) [reply growth/too/tasks/scheduler.py:165]: Sweet!  I'll make that change.

File: growth/too/tasks/scheduler.py [hunk 4]
```diff
@@ -11,9 +11,13 @@
 import astropy.units as u
 from astroplan import Observer, is_always_observable
 from astroplan.constraints import AltitudeConstraint
+from celery.utils.log import get_task_logger
 from . import celery
 from .. import models
 from .. import views
+from flask import flash
```
Comment (lpsinger) [inline growth/too/tasks/scheduler.py:18]: Please collect external imports separately from internal imports.

File: growth/too/tasks/scheduler.py [hunk 5]
```diff
@@ -157,8 +162,12 @@ def schedule_ztf(json_data):
                            'validity_window_mjd':
                                json_data["validity_window_mjd"],
                            'queue_type': 'list'})
-    r.raise_for_status()
 
+    if (r.status_code == 200) or (r.status_code == 400):
```
Comment (lpsinger) [inline growth/too/tasks/scheduler.py:166]: Wait a second. Why `danger` if `r.status_code == 200`?

File: growth/too/tasks/scheduler.py [hunk 6]
```diff
@@ -157,8 +162,12 @@ def schedule_ztf(json_data):
...
+        flash(r.text, 'danger')
+    elif r.status_code == 201:
+        flash(r.text, 'success')
+    log.exception(r.text)
```
Comment (lpsinger) [inline growth/too/tasks/scheduler.py:170]: `log.exception()` should be used inside an `except:` block.

Comment (mcoughlin) [reply growth/too/tasks/scheduler.py:170]: I don't quite understand... what should this be?

File: growth/too/tasks/scheduler.py (hunk 5 above)
Comment (mcoughlin) [reply growth/too/tasks/scheduler.py:166]: This is the case where we resubmitted the same queue twice. It isn't a success.

Comment (lpsinger) [reply growth/too/tasks/scheduler.py:166]: HTTP 200 means "OK".

Comment (mcoughlin) [reply growth/too/tasks/scheduler.py:166]: Ok well I am just interpreting what we are getting back from the scheduler. @ebellm

File: growth/too/tasks/scheduler.py (hunk 6 above)
Comment (lpsinger) [reply growth/too/tasks/scheduler.py:170]:
```python
try:
    r.raise_for_status()
except HTTPError:
    log.exception('submission to scheduler failed (%s)', r.text)
    flash(r.text)
else:
    flash(r.text, success)
```

**Comment by lpsinger** [comment]
You have a lint error here.

**Comment by lpsinger** [comment]
The HTTPError needs to be imported.

**Pull request merged**